connector/mqtt/vernemq/broker/vmq_ssl.erl

```diff
@@ -0,0 +1,192 @@
+%% Copyright 2018 Erlio GmbH Basel Switzerland (http://erl.io)
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%%     http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+
+-module(vmq_ssl).
+-include_lib("public_key/include/public_key.hrl").
+-export([socket_to_common_name/1,
+         cert_to_common_name/1,
+         opts/1]).
+
+check_fingerprint_username(Username,UsernameRedis) ->
+    if
+        Username == UsernameRedis ->
+            Username;
+        true ->
+            not_authorized
+    end.
+
+socket_to_common_name(Socket) ->
+    case ssl:peercert(Socket) of
+        {error, no_peercert} ->
+            undefined;
+        {ok, Cert} ->
+            OTPCert = public_key:pkix_decode_cert(Cert, otp),
+            TBSCert = OTPCert#'OTPCertificate'.tbsCertificate,
+            Subject = TBSCert#'OTPTBSCertificate'.subject,
+            
+            FingerPrint = crypto:hash(sha256, Cert),
+            % the fingerprint at this point is with formatted but with the ":" as the last character
+            FingerPrintFormatted = lists:flatten([io_lib:format("~2.16.0B:",[X]) || <<X:8>> <= FingerPrint ]),
+            ValidFingerPrint = [string:strip(FingerPrintFormatted, right, $:)],
+
+            Username = extract_cn(Subject), 
+            {ok, Client} = eredis:start_link("vernemq-redis", 6379,0),            
+            {ok, ValueReturn} = eredis:q(Client, ["GET", ValidFingerPrint]),
+            eredis:stop(Client),
+           
+            OkReturn = check_fingerprint_username(Username, ValueReturn),
```

Comment:
The Username and ValueReturn are not necessary equals. We removed the rule that the certificate CNAME must meet the pattern `tenant:deviceId`. I think it is only necessary to check if the fingerprint exists in the Redis database.